react/UploadQueue/index.jsx

```diff
@@ -0,0 +1,264 @@
+  id_ID: localeId_ID,
+  it: localeIt,
+  ja: localeJa,
+  ko: localeKo,
+  nl: localeNl,
+  nl_NL: localeNl_NL,
+  pl: localePl,
+  pl_PL: localePl_PL,
+  ru: localeRu,
+  zh_CN: localeZh_CN,
+  zh_TW: localeZh_TW
+}
+
+const CANCEL = 'cancel'
+const PENDING = 'pending'
+const LOADING = 'loading'
+const CREATED = 'created'
+const UPDATED = 'updated'
+const FAILED = 'failed'
+const CONFLICT = 'conflict'
+const QUOTA = 'quota'
+const NETWORK = 'network'
+const DONE_STATUSES = [CREATED, UPDATED]
+const ERROR_STATUSES = [CONFLICT, NETWORK, QUOTA]
+
+const splitFilename = filename => {
+  const parts = filename.split('.')
+  return {
+    filename: parts.slice(0, -1).join('.') + '.',
+    extension: parts[parts.length - 1]
+  }
+}
+
+export const uploadStatus = {
+  CANCEL,
+  PENDING,
+  LOADING,
+  CREATED,
+  UPDATED,
+  FAILED,
+  CONFLICT,
+  QUOTA,
+  NETWORK,
+  DONE_STATUSES,
+  ERROR_STATUSES
+}
+
+const Pending = translate()(props => (
+  <span className={styles['item-pending']}>{props.t('item.pending')}</span>
+))
+
+const FileUploadProgress = ({ progress }) => {
+  return <ThresholdBar threshold={progress.total} value={progress.loaded} />
+}
+
+const Item = translate()(
+  ({ file, status, isDirectory, progress, getMimeTypeIcon }) => {
+    console.log('item', { file, status, progress })
+    const { CANCEL, LOADING, DONE_STATUSES, ERROR_STATUSES } = uploadStatus
+    const { filename, extension } = splitFilename(file.name)
+    let statusIcon
+    let done = false
+    let error = false
+    /**
+     * Status cames from the Upload Queue, but sometimes we're using
+     * manual upload without using the Upload Queue system but we're still
+     * using the UI component. When this is the case, the file handles on
+     * his own its status.
+     */
+    const statusToUse = file.status ? file.status : status
+
+    if (statusToUse === LOADING) {
+      statusIcon = (
+        <Spinner className="u-ml-half" color={palette['dodgerBlue']} />
+      )
+    } else if (statusToUse === CANCEL) {
+      statusIcon = (
+        <Icon className="u-ml-half" icon="cross" color={palette['monza']} />
+      )
+    } else if (ERROR_STATUSES.includes(statusToUse)) {
+      error = true
+      statusIcon = (
+        <Icon className="u-ml-half" icon="warning" color={palette['monza']} />
+      )
+    } else if (DONE_STATUSES.includes(statusToUse)) {
+      done = true
+      statusIcon = (
+        <Icon className="u-ml-half" icon="check" color={palette['emerald']} />
+      )
+    } else {
+      statusIcon = <Pending />
+    }
+
+    return (
+      <div
+        data-test-id="upload-queue-item"
+        className={classNames(styles['upload-queue-item'], {
+          [styles['upload-queue-item--done']]: done,
+          [styles['upload-queue-item--error']]: error
+        })}
+      >
+        <div
+          className={classNames(
+            styles['item-file'],
+            'u-flex',
+            'u-flex-items-center',
+            'u-p-1'
```

Comment:
Since we have a stylus declaration for this element, can we move the utility classes there as well?